PaymentButtons/src/main/java/com/paypal/android/paymentbuttons/PaymentButton.kt

```diff
@@ -94,9 +115,18 @@ abstract class PaymentButton<C : PaymentButtonColor> @JvmOverloads constructor(
                 }
             }
 
-            val cornerTreatment = when (field) {
-                PaymentButtonShape.ROUNDED, PaymentButtonShape.PILL -> RoundedCornerTreatment()
-                PaymentButtonShape.RECTANGLE -> CutCornerTreatment()
+            val cornerTreatment = if (customCornerRadius != null) {
+                if (customCornerRadius == 0.0f) {
+                    CutCornerTreatment()
+                } else {
+                    RoundedCornerTreatment()
+                }
```

Comment:
This seems duplicate of lines 82-86 - does it need to be set again here?

Reply:
I think this is the set function for the `var shape: PaymentButtonShape`. So the `customCornerRadius` variable is checked to see if a value was set previously.

Is this logic intended behavior? If a merchant first sets `customCornerRadius` > 0 and then sets `shape` as `PaymentButtonShape.RECTANGLE`, the button will still display with rounded corners. Should we always style the button with the last property the merchant sets?

Reply:
It'd be nice if we could make `var shape: PaymentButtonShape` a write-only, virtual property and make `var cornerRadius: Float` the source of truth. Kotlin doesn't seem to have explicit support for [write-only props](https://youtrack.jetbrains.com/issue/KT-6519) though. We may still be able to refactor our logic to eliminate nullability for both properties.

The duplication may be a sign of technical debt. It may make sense to the revisit the requirements for these buttons to make sure we're properly aligned with product.